core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java

```diff
@@ -433,17 +438,19 @@ private Resource updateDeleteVolumesFlag(boolean deleteVolumes, Resource volumeS
         return volumeSet;
     }
 
-    public void markVolumesToNonDeletable(StackView stack, Set<String> instanceIdsToRepair) {
-        Predicate<Resource> updateVolumesPredicate = inInstances(instanceIdsToRepair);
-        updateVolumesDeleteFlag(stack, updateVolumesPredicate, false);
+    public void markVolumesToNonDeletable(StackView stack, List<InstanceMetadataView> instanceMetadataViews) {
+        Set<String> instanceIds = instanceMetadataViews.stream().map(InstanceMetadataView::getInstanceId).collect(toSet());
+        Set<String> instanceFQDNs = instanceMetadataViews.stream().map(InstanceMetadataView::getDiscoveryFQDN).collect(toSet());
+        updateVolumesDeleteFlag(stack, instanceIds, instanceFQDNs, false);
     }
 
     public void setStackStatusAndMarkDeletableVolumes(ManualClusterRepairMode repairMode, boolean deleteVolumes, StackView stack,
-            Set<String> instanceIdsToRepair) {
-        if (!ManualClusterRepairMode.DRY_RUN.equals(repairMode) && !instanceIdsToRepair.isEmpty()) {
+            Set<InstanceMetaData> instances) {
+        if (!ManualClusterRepairMode.DRY_RUN.equals(repairMode) && !instances.isEmpty()) {
             LOGGER.info("Repair mode is not a dry run, {}", repairMode);
-            Predicate<Resource> updateVolumesPredicate = inInstances(instanceIdsToRepair);
-            updateVolumesDeleteFlag(stack, updateVolumesPredicate, deleteVolumes);
+            Set<String> instanceIds = instances.stream().map(InstanceMetadataView::getInstanceId).collect(toSet());
+            Set<String> instanceFQDNs = instances.stream().map(InstanceMetadataView::getDiscoveryFQDN).collect(toSet());
+            updateVolumesDeleteFlag(stack, instanceIds, instanceFQDNs, deleteVolumes);
```

Comment:
the logic here is almost the same as the logic in `markVolumesToNonDeletable`, can be extracted both into a method

Reply:
you are right, I will extract it